Sources/Instrumentation/Instrument.swift

```diff
@@ -0,0 +1,51 @@
+import BaggageContext
+
+/// Conforming types are usually cross-cutting tools like tracers that extract values of one type (`ExtractFrom`) into a
+/// `BaggageContext` and inject values stored in a `BaggageContext` into another type (`InjectInto`).
+/// `ExtractFrom` and `InjectInto` may very well be of the same type, e.g. for a tracer that propagates baggage
+/// through HTTP headers.
+public protocol InstrumentProtocol {
+    associatedtype InjectInto
+    associatedtype ExtractFrom
+
+    /// Using this instrument, extract values from an `ExtractFrom` and inject them into the given
```

Comment:
Maybe skip the Using this instrument?

```suggestion
    /// Extract values from an `ExtractFrom` and inject them into the given
```